### DISCUSSION TITLE
Evaluation at cast time?

### DISCUSSION THREAD

**Discussion by ljunio** [colinhacks/zod#919] [Q&A]
Hi there,

Is there a preferred schema or method that I can use to evaluate the type of a data entry at cast time? Yup has the lazy: https://github.com/jquense/yup#lazyvalue-any--schema-lazy function that allows validation on cast time which then outputs type A or type B basically.

Is there something similar for zod?

**Comment by JacobWeisenburger** [comment]
https://github.com/colinhacks/zod#recursive-types

Please let me know if you still have questions.
